### ISSUE TITLE
selective filtering of "too many requests from ip, so ban that ip"

### ISSUE THREAD

**Issue by acertain** [SimpleServer/SimpleServer#168]
on subnet, just dont do it, or ley us add ips to a file that will not be affected by this...or even option to disable

**Comment by pimanac** [comment]
Agreed on option to disable or set the sensitivity (i.e number of requests per minute or something).  Does anyone know what could cause too many requests from a client?  Is it too many login requests, could the clients be acting up?  Flakey internet connection?  Thanks

**Comment by acertain** [comment]
hitting the refresh button too many times...

**Comment by pimanac** [comment]
I can see where that would cause this issue, but I am 99.9% sure that the players connecting to my server are not hitting the refresh button 30 times in a minute.

any other thoughts?

**Comment by acertain** [comment]
its like 10 times/min that does it

**Comment by pimanac** [comment]
according to the documentation on the main page:
- Denial of service protection.  More than 30 requests/IP/minute will
  automatically ban the IP.

so clicking the refresh initiates three separate requests?

**Comment by acertain** [comment]
no, but it does for me bcuz i have it in my server browser as 3 interfaces

**Comment by Raptor2213** [comment]
Banning the IP wouldn't do anything anyway.  Even a banned IP will still be able to query the server, just not connect to the server.

**Comment by acertain** [comment]
false.

**Comment by The-PC-Tech-Guy** [comment]
A banned IP is indeed allowed to query the server, they just cannot join it. But it is true that I'd like to customize this "feature" due to the new server list present in the client (this feature was made way back in the alpha days, if I'm not mistaken).

**Comment by acertain** [comment]
thats what I get...but I am making a ridiculous # of requests.

**Comment by The-PC-Tech-Guy** [comment]
Well, this one isn't bad, because it involves a change in Minecraft. And, I have experienced it as well - one IP got banned on my server, and I didn't know why until I found out it was making so many requests.
